src/main/java/software/amazon/awssdk/crt/io/TlsKeyOperation.java

```diff
@@ -0,0 +1,226 @@
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+package software.amazon.awssdk.crt.io;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import software.amazon.awssdk.crt.Log;
+import software.amazon.awssdk.crt.Log.LogLevel;
+import software.amazon.awssdk.crt.Log.LogSubject;
+
+import software.amazon.awssdk.crt.CrtResource;
+
+/**
+ * A class containing a mutual TLS (mTLS) Private Key operation that needs to be performed.
+ * This class is passed to TlsKeyOperationHandler if a custom key operation is set.
+ *
+ * You MUST call either complete(output) or completeExceptionally(exception)
+ * or the TLS connection will hang forever!
+ *
+ * You do NOT need to call close on this CrtResource - it will be called automatically
+ * when you call either complete(output) or completeExceptionally(exception)
+ */
+public final class TlsKeyOperation extends CrtResource {
```

Comment:
I'm worried about this being a CrtResource, it gives users more functions that they might call incorrectly (`close()`, `acquireRef()`)

This function already has its own special lifetime rules, that one of the complete() functions must be called

how about we make this a normal class

but store the nativeHandle as some private member, which still IS a CrtResource (so we get leak tracking)